pkg/testing/http_mock.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const (
+	OctetStream = "application/octet-stream"
+	JSON        = "application/json; charset=utf-8"
+	TEXT        = "text/plain"
+)
+
+var DefaultRoundTripper http.RoundTripper = &http.Transport{
```

Comment:
Did you consider using https://golang.org/pkg/net/http/httptest/ instead of mocking at this level?

Reply:
Looking into it.